yarn-project/aztec/src/cli/cmds/start_prover_agent.ts

```diff
@@ -1,62 +1,55 @@
 import { BBNativeRollupProver, TestCircuitProver } from '@aztec/bb-prover';
 import { type ServerCircuitProver } from '@aztec/circuit-types';
-import { getProverEnvVars } from '@aztec/prover-client';
+import { ProverClientConfig, proverClientConfigMappings } from '@aztec/prover-client';
 import { ProverAgent, createProvingJobSourceClient } from '@aztec/prover-client/prover-agent';
 import {
+  TelemetryClientConfig,
   createAndStartTelemetryClient,
-  getConfigEnvVars as getTelemetryClientConfig,
+  telemetryClientConfigMappings,
 } from '@aztec/telemetry-client/start';
 
-import { type ServiceStarter, parseModuleOptions } from '../util.js';
+import { type ServiceStarter, extractRelevantOptions } from '../util.js';
 
 export const startProverAgent: ServiceStarter = async (options, signalHandlers, logger) => {
-  const proverOptions = {
-    ...getProverEnvVars(),
-    ...parseModuleOptions(options.prover),
-  };
+  const proverConfig = extractRelevantOptions<ProverClientConfig>(options, proverClientConfigMappings);
 
-  if (!proverOptions.nodeUrl) {
+  if (!proverConfig.nodeUrl) {
     throw new Error('Starting prover without an orchestrator is not supported');
   }
 
-  logger(`Connecting to prover at ${proverOptions.nodeUrl}`);
-  const source = createProvingJobSourceClient(proverOptions.nodeUrl, 'provingJobSource');
+  logger(`Connecting to prover at ${proverConfig.nodeUrl}`);
+  const source = createProvingJobSourceClient(proverConfig.nodeUrl, 'provingJobSource');
 
-  const agentConcurrency =
-    // string if it was set as a CLI option, ie start --prover proverAgentConcurrency=10
-    typeof proverOptions.proverAgentConcurrency === 'string'
-      ? parseInt(proverOptions.proverAgentConcurrency, 10)
-      : proverOptions.proverAgentConcurrency;
+  // NOTE: Shouldn't telemetry be optional?
```

Comment:
It is, it defaults to a no-op client if not configured.

Reply:
meaning we just leave `telemetryConfig` empty?

Reply:
ah nvm I see now